static/js/modules/bar-charts.js

```diff
@@ -0,0 +1,269 @@
+    })
+    .select('tspan:nth-child(2)')
+    .attr('x', 0)
+    .attr('dy', '1.5em');
+
+  var periods = svg.selectAll('.period')
+      .data(data)
+    .enter()
+      .append('g')
+      .attr('class', function(d) { return 'period ' + d.status; })
+      .attr('transform', function(d) { return 'translate(' + x0(d.period) + ',0)'; })
+      .attr('tabindex', '0')
+      .on('focus', this.showTooltip.bind(this, x0))
+      .on('blur', this.hideTooltip.bind(this))
+      .on('mouseenter', this.showTooltip.bind(this, x0))
+      .on('mouseleave', this.hideTooltip.bind(this));
+
+  var periodsWithData = periods.filter(function(d) {
+   return d.status === 'complete' || d.status === 'in-progress';
+  });
+
+  var emptyperiods = periods.filter(function(d) {
+    return d.status === 'not-started';
+  });
+
+  // Assign to a variable so it can be used in callbacks
+  var height = this.height;
+
+  periodsWithData
+      .selectAll('.bar')
+      .data(function(d) {
+        return d.entities;
+      })
+    .enter()
+      .append('rect')
+      .attr('class', function(d) {
+        return 'bar ' + d.name;
+      })
+      .attr('y', height)
+      .attr('height', 0)
+      .attr('width', x1.rangeBand())
+      .attr('x', function(d) { return x1(d.name); })
+      .transition()
+        .duration(1000)
+        .attr('height', function(d) {
+          var barHeight = height - y(d.value);
+          return barHeight;
+        })
+        .attr('y', function(d) { return y(d.value); });
+
+  periodsWithData.filter(function(d) { return d.status === 'in-progress'; })
+    .selectAll('.bar--in-progress')
+    .data(function(d) {
+      return d.entities;
+    })
+    .enter()
+      .append('rect')
+      .attr('class', 'bar--in-progress')
+      .attr('y', height)
+      .attr('height', 0)
+      .attr('width', x1.rangeBand())
+      .attr('x', function(d) { return x1(d.name); })
+      .transition()
+        .duration(1000)
+        .attr('height', function(d) {
+          var barHeight = height - y(d.value);
+          return barHeight;
+        })
+        .attr('y', function(d) { return y(d.value); });
+
+  periodsWithData.insert('rect', '.bar')
+    .attr('class', 'bar-bg')
+    .attr('width', x1.rangeBand() * 4)
+    .attr('height', this.height)
+    .attr('fill', 'transparent')
+    .style('cursor', 'pointer');
+
+  emptyperiods.selectAll('rect')
+      .data(data)
+      .enter()
+        .append('rect')
+        .attr('class', 'bar--empty')
+        .attr('width', x1.rangeBand() * 4)
+        .attr('x', function(d) { return x1(d.name); })
+        .attr('y', 0)
+        .attr('height', this.height);
+
+  return svg;
+};
+
+GroupedBarChart.prototype.appendTooltip = function() {
+  return this.element
+    .style({'position': 'relative'})
+    .append('div')
+    .attr('class', 'tooltip tooltip--under tooltip--chart')
+    .attr('id', 'tooltip');
+};
+
+GroupedBarChart.prototype.populateTooltip = function(d) {
+  var entityMap = {
```

Comment:
Not a big deal, but when mapping internal names to UI names, I prefer to use the term "display names" so it's clear that these are meant for display to the user, not meant for other computation. I might consider naming this "entityDisplayNamesMap". I know, verbose :)